pipelines/controller/src/main/java/org/openmrs/analytics/HiveTableManager.java

```diff
@@ -76,26 +75,75 @@ public void createResourceTables(
   private void createResourceTable(
       Connection connection, String resource, String timestamp, String thriftServerParquetPath)
       throws SQLException {
-    try (Statement statement = connection.createStatement()) {
-      String sql =
-          String.format(
-              "CREATE TABLE IF NOT EXISTS default.%s_%s USING PARQUET LOCATION '%s/%s/%s'",
-              resource,
-              timestamp,
-              THRIFT_CONTAINER_PARQUET_PATH_PREFIX,
-              thriftServerParquetPath,
-              resource);
-      statement.execute(sql);
 
-      // Drop canonical table if exists.
-      sql = String.format("DROP TABLE IF EXISTS default.%s", resource);
-      statement.execute(sql);
+    String sql =
+        String.format(
+            "CREATE TABLE IF NOT EXISTS default.%s_%s USING PARQUET LOCATION '%s/%s/%s'",
+            resource,
+            timestamp,
+            Constants.THRIFT_CONTAINER_PARQUET_PATH_PREFIX,
+            thriftServerParquetPath,
+            resource);
+    createResourceTable(connection, sql);
+
+    // Drop canonical table if exists.
+    sql = String.format("DROP TABLE IF EXISTS default.%s", resource);
+    createResourceTable(connection, sql);
```

Comment:
Can you check if thrift-server (with default Derby) supports transactions? If yes, then let's try to make a single transaction for this and the next statement such that we never end up in a state were a canonical table is deleted and the next one is not created yet.

Reply:
According to my findings, Hive supports transactional tables:
https://docs.cloudera.com/HDPDocuments/HDP3/HDP-3.0.1/using-hiveql/content/hive_3_internals.html

where you can create tables like this:
`CREATE TABLE tm (a int, b int) TBLPROPERTIES
('transactional'='true', 'transactional_properties'='insert_only')`

I dont think we can achieve the aforementioned transaction here.

Reply:
So to clarify what I meant: I was proposing that we bundle this `DROP` and the next `CREATE` table queries into a single transaction by [`setAutoCommit`](https://docs.oracle.com/en/java/javase/17/docs/api/java.sql/java/sql/Connection.html#setAutoCommit(boolean)) to false and then call `conneciton.commit()`. This way either both of the queries succeed or both fail. Are you saying this fails with the default Metastore?

Reply:
I tried the suggested approach and got this warning:

`pipeline-controller    | 21:00:11.818 [Thread-1] WARN  org.apache.hive.jdbc.HiveConnection org.apache.hive.jdbc.HiveConnection.setAutoCommit:1370 - Request to set autoCommit to false; Hive does not support autoCommit=false.`